src/app/SERVICE/tasks.service.ts

```diff
@@ -1,48 +1,50 @@
 import { Injectable } from '@angular/core';
 import { Task } from '../models/task.model';
-import { BehaviorSubject, Observable, map } from 'rxjs';
+import { Observable, map, switchMap } from 'rxjs';
 import { HttpClient } from '@angular/common/http';
 import { Router } from '@angular/router';
-import { Constants } from '../utils/constants';
+import { tasksApiUrl } from '../utils/constants';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TasksService {
 
-  constructor(private http: HttpClient, private router: Router, private constants: Constants) {
+  constructor(private http: HttpClient, private router: Router) {
```

Comment:
It doesn't seem like you're using router in this service, you can remove it if that's the case